cartridge/utils.lua

```diff
@@ -356,13 +356,21 @@ local function http_read_body(req)
     -- that have a special meaning with % to escape them.
     -- A list of special characters is ().+-*?[]^$%
     local boundary_line = string.gsub('--'..boundary, "[%(%)%.%+%-%*%?%[%]%^%$%%]", "%%%1")
-    local _, form_body = req_body:match(
+    local payload_headers, form_body = req_body:match(
         boundary_line .. '\r\n' ..
         '(.-\r\n)' .. '\r\n' .. -- headers
         '(.-)' .. '\r\n' .. -- body
         boundary_line
     )
-    return form_body
+
+    local filename, _
+    if payload_headers ~= nil then
+        _, filename = payload_headers:match('Content%-Disposition%: form%-data; name="(.-)"; filename="([^"]-)"')
+    end
+
+    return form_body, nil, {
+        filename = filename,
```

Comment:
Why do you return filename inside table?

Reply:
It's an approach that allows to extend API. Probably some other headers will be also required and it will be possible to add just new field to a table.